src/features/treasureIsland/components/TreasureTrove.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+
+import { MapPlacement } from "features/game/expansion/components/MapPlacement";
+
+import { Modal } from "react-bootstrap";
+import { CloseButtonPanel } from "features/game/components/CloseablePanel";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { getEntries } from "features/game/types/craftables";
+import { Label } from "components/ui/Label";
+import { SUNNYSIDE } from "assets/sunnyside";
+import { secondsToString } from "lib/utils/time";
+import { TREASURES } from "features/game/types/treasure";
+import { NPC } from "features/island/bumpkin/components/DynamicMiniNFT";
+import { Equipped } from "features/game/types/bumpkin";
+
+enum RarityOrder {
+  "rare",
+  "good",
+  "average",
+}
+
+const TREASURE_TROVE_ITEMS = getEntries(TREASURES)
+  // Sort by name first
+  .sort(([nameA], [nameB]) => nameA.localeCompare(nameB))
+  // Then sort by rarity
+  .sort(
+    ([, treasureA], [, treasureB]) =>
+      RarityOrder[treasureA.type] - RarityOrder[treasureB.type]
+  );
+
+const LIMITED_TREASURE_TROVE_ITEMS = TREASURE_TROVE_ITEMS.filter(
+  ([, treasure]) => !!treasure.endDate && treasure.endDate >= Date.now()
+);
+
+const UNLIMITED_TREASURE_TROVE_ITEMS = TREASURE_TROVE_ITEMS.filter(
+  ([, treasure]) => !treasure.endDate
+);
+
+const TreasureTroveItem: React.FC<{
+  treasureName: keyof typeof TREASURES;
+  rarity: "good" | "average" | "rare";
+  endDate?: number;
+}> = ({ treasureName, rarity, endDate }) => (
+  <div key={treasureName} className="flex justify-between">
```

Comment:
I think there could be an improvement on the layout of the limited item. What do you think of this:
```suggestion
    <>
    {endDate && (
      <div className="flex items-start justify-between">
        <span className="text-xs ml-2 italic">Time Limited Treasure!</span>
        <Label type="info" className="flex items-center whitespace-nowrap">
          <img src={SUNNYSIDE.icons.stopwatch} className="w-3 left-0 mr-1" />
          {`${secondsToString((endDate - Date.now()) / 1000, {
            length: "medium",
            isShortFormat: true,
          })} left`}
        </Label>
      </div>
    )}
    <div key={treasureName} className="flex justify-between items-center">
      <div className="flex-col space-y-1">
        <div className="flex">
          <div className="w-12 justify-center items-center flex mr-2">
            <img src={ITEM_DETAILS[treasureName].image} className="h-8" />
          </div>
          <div className="justify-center items-center flex">
            <span className="text-sm mb-1">{treasureName}</span>
          </div>
        </div>
      </div>

      <div className="flex items-center">
        {rarity === "rare" && <Label type="warning">Rare</Label>}
        {rarity === "good" && <Label type="info">Uncommon</Label>}
        {rarity === "average" && <Label type="success">Common</Label>}
      </div>
    </div>
  </>
```
<img width="376" alt="Screen Shot 2023-02-02 at 1 01 22 pm" src="https://user-images.githubusercontent.com/17863697/216212879-96dda694-cba5-4378-ba76-8cb4a5dcfa1b.png">